src/main/java/Duke.java

```diff
@@ -1,10 +1,218 @@
+import exception.EmptyTaskDescException;
+import exception.EmptyTaskTimeException;
+import exception.NoSuchTaskException;
+import tasks.*;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.Scanner;
+
 public class Duke {
+
+    public enum TaskTypes {
+        TODO,
+        DEADLINE,
+        EVENT
+    }
+
+    private List<Task> taskList;
+    private SaveData data;
+    private static final File savefile = new File("savedata.txt");
+
     public static void main(String[] args) {
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+
+        Duke duke = new Duke();
+        duke.start();
+    }
+
+    public Duke() {
+
+    }
+
+    public void start() {
+        Scanner sc = new Scanner(System.in);
+        System.out.println(
+                "Hello! I'm Duke\n" +
+                "What can I do for you?"
+        );
+        data = new SaveData();
+        boolean isSaveFileCreated = savefile.exists();
+        if (!isSaveFileCreated) {
+            try {
+                isSaveFileCreated = savefile.createNewFile();
+            } catch (IOException e) {
+                System.out.println("Error while creating save file: " + e);
+            }
+        }
+
+        if (!isSaveFileCreated) {
+            System.out.println("Unable to find or create save file, exiting...");
+            System.exit(1);
+        }
+
+        data.loadFromFile(new File("savedata.txt"));
+        taskList = data.getTaskList();
+        loop(sc);
+    }
+
+    public void loop(Scanner sc) {
```

Comment:
This method looks long and complex, could think of abstracting out key components into different methods?